### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by KUBAHO3**
**In this Project we have:**

- Insert and query data in SQL.
- Use primary key & foreign key mechanisms for joining tables.
- Understand the different types of relationships between tables.
- Query multiple tables.
- Prepare complex queries that answer analytical questions.
- Run a program using the command line.
- Use Ruby syntax for basic programming operations.
- Apply Ruby best practices and language style guides in code.
- Describe the SOLID principles of OOP.
- Implement classes and objects in Ruby.
- Understand the four main principles of OOP.
- Implement encapsulation and inheritance with Ruby.
- Create a UML class diagram.
- Explain the difference between associations, aggregations, and composition in OOP.
- Set up associations between classes and objects.
- Store data in files.
- Build interactive console apps.
- Recognize the value of making equal contributions to group projects to produce the best outcome.

### CODE REVIEW

File: item.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def move_to_archive
+    self.archived = can_be_archived?
+  end
+
+  def add_genre(genre)
+    @genre = genre
+    genre.items.push(self) unless genre.items.include?(self)
+  end
```
Comment (tresorsawasawa) [inline item.rb:25]: - [ ] As required for the project and to correctly match the relationship between classes, kindly ensure that the `Item` class has one `Genre`, `Author`, `Label`. Right now, I noticed that you miss both the `Author` and `Label` in the `Item` class. Make sure you add them by following the same logic you use in the `add_genre(genre)` method and `Getter` should be defined to each one(i.e. ``attr_reader ... :genre, :author, :label`` )

File: item.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  def can_be_archived?
+    year = Time.new.year
+    year - @publish_date.year > 10
+  end
+
+  def move_to_archive
+    self.archived = can_be_archived?
+  end
+
```
Comment (tresorsawasawa) [inline item.rb:21]: - [ ] As required for the project, I noticed that the `move_to_archive()` method in the Item class has to reuse `can_be_archived?()` method, to change the archived property to true if the result of the` can_be_archived?()` method is true, and do nothing if the result of the `can_be_archived?()` method is false. But in your case, I can't see all the conditions be implemented. I think this should be updated to something like this:
```ruby
   def move_to_archive
      if can_be_archived?
        @archived = true
      end
    end
```

File: spec/genre_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  describe '#items' do
+    it 'should return the item1 of the Genre' do
+      expect(@genre.items[0]).to eq 'item1'
+    end
+    it 'should return the item2 of the Genre' do
+      expect(@genre.items[1]).to eq 'item2'
+    end
+  end
+end
```
Comment (tresorsawasawa) [inline spec/genre_spec.rb:28]: - [ ] As required for the project, all public methods in all classes to have unit tests, kindly ensure that you also add tests to the `add_item()` method in the `Genre` class.

File: spec/label_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  describe '#add_label' do
+    it 'adds a book to the label items' do
+      label = Label.new('Title', 'Color')
+      book = double('Book')
+      label.add_label(book)
+      expect(label.items).to include(book)
+    end
+  end
```
Comment (tresorsawasawa) [inline spec/label_spec.rb:21]: - [ ] As required for the project, all public methods in all classes to have unit tests, kindly ensure that you also add tests to the `add_item()` method in the `Label` class after it has been implemented

File: item.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+require 'date'
+
+class Item
```
Comment (tresorsawasawa) [inline item.rb:3]: - [ ] As required for the project, all public methods in all classes to have unit tests, kindly ensure that you add tests to the `Item` class and perform tests to the public method(s) inside it.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @KUBAHO3, @joyapisi, and @BANCUNGUYE66 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [x] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KUBAHO3** [comment]
Thank you @tresorsawasawa for all the requested changes! We have worked on it and now everything is running fine!

**Review by franclobo — APPROVED** [review]
Hi @KUBAHO3, @joyapisi, and @BANCUNGUYE66, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️
- Descriptive PR ✔️
- Correct Git flow ✔️
- Use OOP ✔️
- All tests passed ✔️
- Store data in files ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**